library/java/src/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -1,11 +1,38 @@
 package io.envoyproxy.envoymobile.engine;
 
-public interface EnvoyConfiguration {
-
-  /**
-   * Provides a default configuration template that may be used for starting Envoy.
-   *
-   * @return A template that may be used as a starting point for constructing configurations.
-   */
-  String templateString();
+public class EnvoyConfiguration {
+
+  public final String configYAML;
+  public final int connectTimeoutSeconds;
+  public final int dnsRefreshSeconds;
+  public final int statsFlushSeconds;
+
+  public EnvoyConfiguration(String configYAML, int connectTimeoutSeconds, int dnsRefreshSeconds,
+                            int statsFlushSeconds) {
+    this.configYAML = configYAML;
+    this.connectTimeoutSeconds = connectTimeoutSeconds;
+    this.dnsRefreshSeconds = dnsRefreshSeconds;
+    this.statsFlushSeconds = statsFlushSeconds;
+  }
+
+  public String resolve(String defaultConfigurationYAML) {
```

Comment:
+1 for making this private/internal.

Additionally, `resolveTemplate(String templateYAML)` would make sense IMO.

One other thing that I think we should do here is not check `configYAML` within this function, and instead have it simply resolve the `templateYAML` it's passed. The caller can then be responsible for determining whether to use a pre-resolved YAML string or if it needs to resolve a configuration against a template. This would reduce the amount of magic done in this function, especially if we take the approach suggested above which treats user-specified configurations as pre-resolved (not resolving them as templates)